plugins/ruby/plugin.zsh

```diff
@@ -0,0 +1,68 @@
+# Color definitions
+GEOMETRY_COLOR_RUBY_RVM_VERSION=${GEOMETRY_COLOR_PROMPT:-white}
+
+GEOMETRY_RUBY_RVM_SHOW_GEMSET=${GEOMETRY_RUBY_RVM_SHOW_GEMSET:-true}
+
+# Symbol definitions
+GEOMETRY_SYMBOL_RUBY_RVM_VERSION=${GEOMETRY_SYMBOL_RUBY_RVM_VERSION:-"◆"}
+GEOMETRY_RUBY_RVM_VERSION=$(prompt_geometry_colorize $GEOMETRY_COLOR_RUBY_RVM_VERSION $GEOMETRY_SYMBOL_RUBY_RVM_VERSION) 
+
+prompt_geometry_get_full_ruby_version() {
+  if (( $+commands[ruby] )); then
+      GEOMETRY_RUBY_VERSION_FULL="$(ruby -v)"
+  fi
+}
+
+prompt_geometry_get_full_rvm_version() {
+  if (( $+commands[rvm] )); then
+      GEOMETRY_RVM_VERSION_FULL="$(rvm -v)"
+  fi
+}
+
+geometry_prompt_ruby_setup() {}
+
+prompt_geometry_current_rvm_gemset_name() {
+  if $GEOMETRY_RUBY_RVM_SHOW_GEMSET; then
+      local cur_dir=$(pwd)
+      local gemset_name=$(rvm current)
+      [[ $gemset_name =~ 'ruby-[0-9.]+@?(.*)' ]]
+
+      # If no name present, then it's the default gemset
+      if [[ -z $match[1] ]]; then
+          echo "default"
+      else
+          echo $match[1]
+      fi
+  fi
+}
+
+geometry_prompt_ruby_render() {
+  if (( ! $+commands[ruby] )); then
+      return "";
+  fi
+
+  prompt_geometry_get_full_ruby_version
+  [[ $GEOMETRY_RUBY_VERSION_FULL =~ 'ruby ([0-9a-zA-Z.]+)' ]]
+  GEOMETRY_RUBY_VERSION=$match[1]
```

Comment:
These two lines could be moved into a function such as 'prompt_geometry_ruby_version'. Same for rvm.